src/components/MediaDevices/SpeakerCheck.tsx

```diff
@@ -0,0 +1,162 @@
+/// <reference types="vite/client" />
+// needed to load the .wav file
+
+import React, { useEffect, useRef, useState } from 'react';
+import { useDevices } from '@daily-co/daily-react';
+import { Link } from 'react-router-dom';
+
+import { AudioAnalyser } from '../shared/AudioVisualiser/AudioAnalyser.ts';
+import { AudioVisualiser } from '../shared/AudioVisualiser/AudioVisualiser.tsx';
+import { Card } from '../shared/Card/Card';
+import { TroubleShooting } from '../shared/TroubleShooting/TroubleShooting';
+
+import sound from '../shared/AudioVisualiser/sound.wav';
+import { logger } from '../../utils/Logger';
+
+export const SpeakerCheck: React.FC = () => {
+  const { speakers, setSpeaker, currentSpeaker } = useDevices();
+
+  const audioRef = useRef<HTMLAudioElement>(null);
+  const [isPlaying, setIsPlaying] = useState(false);
+  const [speakerAnalyser, setSpeakerAnalyser] = useState<AudioAnalyser | null>(
+    null,
+  );
+
+  const [showTroubleshooting, setShowTroubleshooting] = useState(false);
+  const toggleTroubleShooting = () => {
+    if (showTroubleshooting) {
+      setShowTroubleshooting(false);
+    } else {
+      setShowTroubleshooting(true);
+    }
+  };
+
+  useEffect(() => {
+    const audio = audioRef.current;
+    const handlePlaying = () => {
+      setIsPlaying(true);
+    };
+    const handleEnded = () => {
+      setIsPlaying(false);
+    };
+    if (audio) {
+      audio.addEventListener('playing', handlePlaying);
+      audio.addEventListener('ended', handleEnded);
```

Comment:
One more wild idea: what if we passed these as `onPlaying` and `onEnded` JSX props to the `<audio>` element? Than we could remove this `useEffect`, too 😁

Reply:
The reason we added this in the first place was an issue with... ugh I can't remember. The audio not picking up after switching devices in a particular way? I'll see if everything still works when I remove this!